### PR TITLE
Add unsigned char cast ifdef to work around `-Wtype-limits` warnings

### PR DESCRIPTION
**Pull request by bruvzg**
Suppresses build warnings on ARM Linux.

### CODE REVIEW

File: core/string/ustring.cpp [hunk 1]
```diff
@@ -544,7 +552,11 @@ String &String::operator+=(const char *p_str) {
 	char32_t *dst = ptrw() + lhs_len;
 
 	for (size_t i = 0; i <= rhs_len; i++) {
+#if CHAR_MIN == 0
+		uint8_t c = p_cstr[i];
```
Comment (akien-mga) [inline core/string/ustring.cpp:556]:
```suggestion
		uint8_t c = p_str[i];
```

**Review by akien-mga — APPROVED** [review]
Tested on a Raspberry Pi 5 running Raspberry Pi OS, works fine.

Not a critical problem, but I like my stable releases warning free whenever possible, so this should be good to merge for 4.2.

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.4.

**Pull request merged**